src/kernel/arch/aarch64/interrupts.zig

```diff
@@ -28,6 +28,108 @@ comptime {
+        \\ bl exceptionHandler
+        \\.balign 0x80
+        \\ bl exceptionHandler
+        \\.balign 0x80
+        \\ bl exceptionHandler
+        \\.balign 0x80
+        \\ bl exceptionHandler
+    );
+}
+
+export fn exceptionHandler() noreturn {
+    const exception_entry_number = @truncate(u32, lr() >> 7 & 0xf);
+    var elr_elx: u64 = undefined;
+    var esr_elx: u64 = undefined;
+    var far_elx: u64 = undefined;
+    var sctlr_elx: u64 = undefined;
+    var spsr_elx: u64 = undefined;
+    var vbar_elx: u64 = undefined;
+    if (currentEl() == 3) {
+        elr_elx = mrsEl("elr_el", 3);
+        esr_elx = mrsEl("esr_el", 3);
+        far_elx = mrsEl("far_el", 3);
+        sctlr_elx = mrsEl("sctlr_el", 3);
+        spsr_elx = mrsEl("spsr_el", 3);
+        vbar_elx = mrsEl("vbar_el", 3);
+    } else if (currentEl() == 2) {
+        elr_elx = mrsEl("elr_el", 2);
+        esr_elx = mrsEl("esr_el", 2);
+        far_elx = mrsEl("far_el", 2);
+        sctlr_elx = mrsEl("sctlr_el", 2);
+        spsr_elx = mrsEl("spsr_el", 2);
+        vbar_elx = mrsEl("vbar_el", 2);
+    } else if (currentEl() == 1) {
+        elr_elx = mrsEl("elr_el", 1);
+        esr_elx = mrsEl("esr_el", 1);
+        far_elx = mrsEl("far_el", 1);
+        sctlr_elx = mrsEl("sctlr_el", 1);
+        spsr_elx = mrsEl("spsr_el", 1);
+        vbar_elx = mrsEl("vbar_el", 1);
+    }
+    log.logError("\n", .{});
+    switch (esr_elx) {
+        0x86000010 => {
+            log.logError("instruction abort\n", .{});
+            log.logError("exception level {} - no change\n", .{ currentEl()});
+            log.logError("32 bit instruction at 0x{x} accessing 0x{x}\n", .{ elr_elx, far_elx });
+        },
+        0x96000021 => {
+            log.logError("alignment fault data abort\n", .{});
+            log.logError("exception level {} - no change\n", .{ currentEl()});
+            log.logError("32 bit instruction at 0x{x} accessing 0x{x}\n", .{ elr_elx, far_elx });
+        },
+        0x96000050 => {
+            log.logError("synchronous external data abort\n", .{});
+            log.logError("exception level {} - no change\n", .{ currentEl()});
+            log.logError("32 bit instruction at 0x{x} accessing 0x{x}\n", .{ elr_elx, far_elx });
+        },
+        else => {
+            log.logError("arm exception taken\n", .{});
+        },
+    }
+    log.logError("current exception level {x} exception entry number 0x{x}\n", .{ currentEl(), exception_entry_number });
+    log.logError("vbar_el{} 0x{x}\n", .{currentEl(), vbar_elx});
+    log.logError("elr_el{} {x}\n", .{currentEl(), elr_elx});
+    log.logError("esr_el{} {x} class 0x{x}\n", .{currentEl(),  esr_elx, esr_elx >> 26 & 0x3f });
+    log.logError("far_el{} {x}\n", .{currentEl(), far_elx});
+    log.logError("sctlr_el{} {x}\n", .{currentEl(), sctlr_elx});
+    log.logError("spsr_el{} {x}\n", .{currentEl(), spsr_elx});
+    log.logError("core 0 is now idle in arm exception handler (other cores were already idle)", .{});
```

Comment:
I think we can leave this message out since it dilutes the output a bit.

Reply:
Ok, will do.

Reply:
Done.